src/runtime/statistics.py

```diff
@@ -138,6 +137,66 @@ def harmonic_mean(x: list[i32]) -> f64:
 
     return k / sum
 
+@overload
+def mode(x: list[i32]) -> i32:
+    """
+    Returns the mode of a data sequence of numbers
+    """
+    k: i32 = len(x)
+    if k == 0:
+        raise Exception("No mode for empty data")
+    i: i32
+    j: i32
+    ans: i32 = x[0]
+    for i in range(k):
+        for j in range(i + 1, k):
+            if x[i] > x[j]:
+                x[i], x[j] = x[j], x[i]
+
```

Comment:
I am not sure that this works in lpython currently, and that is why the CI fails.

Reply:
Well making it work is easy.

1. First check whether the assignment is `TupleConstant` to `TupleConstant`. Assume that both `TupleConstant`s have same type in LLVM backend. Reason - If they weren't having same types then AST to ASR transition should have raised a semantic error.
2. Now, take `deepcopy` of each of the elements present in the value `TupleConstant` and store them in a vector of type, `Vec<llvm::Value*> src_values`.  Now do,

```cpp
ptr_loads_copy = ptr_loads;
ptr_loads = 0;
for( size_t i = 0; i < target_tuple->n_elements; i++ ) {
     visit_expr_wrapper(*target_tuple->m_elements[i], false);
     LLVM::CreateStore(*builder, src_values[i], tmp);
}
ptr_loads = ptr_loads_copy;
```

Do you want to try out working on this @Smit-create?